BackEnd/profiles/views.py

```diff
@@ -264,3 +269,29 @@ class RegionDetail(RetrieveUpdateDestroyAPIView):
     serializer_class = RegionSerializer
     permission_classes = (IsAdminUser,)
     queryset = Region.objects.all()
+
+
+class ProfileModeration(APIView):
+    queryset = Profile.objects.active_only()
+
+    def get_object(self):
+        return get_object_or_404(
+            self.queryset, pk=decode_uid(self.request.data.get("uid"))
+        )
+
+    def post(self, request, *args, **kwargs):
+        profile = self.get_object()
+        serializer = ProfileModerationSerializer(data=request.data)
+
+        manager = ModerationManager(profile)
+
+        if serializer.is_valid(raise_exception=True):
+            if request.data.get("action") == "approve":
+                manager.approve_image()
+
+            return Response(status=status.HTTP_204_NO_CONTENT)
+
+        elif request.data.get("action") == "reject":
+            pass
```

Comment:
it would be better to use HTTP_204_NO_CONTENT in both approve and reject

Reply:
That's exactly how it should have been. But I incorrectly indented the if/elif blocks.
Fixed.